src/scripts/main.js

```diff
@@ -1,3 +1,49 @@
 'use strict';
 
-// write code here
+const sortEmployeesBySalary = function (selector) {
+  // Получаем все элементы списка
+  const employeeItems = document.querySelectorAll(selector);
+
+  // Преобразуем NodeList в массив и сортируем его по зарплате
+  const sortedEmployees = Array.from(employeeItems).sort((a, b) => {
+    // Извлекаем зарплату из атрибутов данных и преобразуем в число
+    const salaryA = parseInt(a.dataset.salary.replace(/[$,]/g, ''), 10);
+    // Удаляем $ и запятые
+    const salaryB = parseInt(b.dataset.salary.replace(/[$,]/g, ''), 10);
+
+    return salaryB - salaryA; // Сортируем по убыванию
+  });
+
+  // Очищаем существующий список и добавляем отсортированные элементы
+  const employeeList = document.querySelector('ul');
+
+  employeeList.innerHTML = ''; // Очищаем существующий список
+
+  // Добавляем отсортированные элементы обратно в список
+  sortedEmployees.forEach((employee) => {
+    employeeList.appendChild(employee); // Добавляем элемент
+  });
+};
```

Comment:
Code Quality Issue: The function `sortEmployeesBySalary` mutates the DOM directly. It's generally a bad practice to manipulate the DOM directly because it can lead to performance issues. Instead, I would recommend to return the sorted array from this function and use it as needed elsewhere in your code.